File: src/main/java/com/hp/jipp/model/IdentifyAction.java
package com.hp.jipp.model;

import com.google.auto.value.AutoValue;
import com.hp.jipp.encoding.Keyword;
import com.hp.jipp.encoding.KeywordType;

@AutoValue
public abstract class IdentifyAction extends Keyword {
    public static final IdentifyAction Display = of("display");
    public static final IdentifyAction Flash = of("flash");
    public static final IdentifyAction Sound = of("sound");
    public static final IdentifyAction Speak = of("speak");

    public static final KeywordType.Encoder<IdentifyAction> ENCODER = KeywordType.Encoder.Companion.of(
            IdentifyAction.class, new Keyword.Factory<IdentifyAction>() {
                @Override
                public IdentifyAction of(String name) {
                    return IdentifyAction.of(name);
                }
            });

    public static KeywordType<IdentifyAction> typeOf(String name) {
        return new KeywordType<>(ENCODER, name);
    }

    public static IdentifyAction of(String name) {
        return new AutoValue_IdentifyAction(name);
    }
}

File: src/main/java/com/hp/jipp/model/IdentifyAction.kt
// Copyright 2017 HP Development Company, L.P.
// SPDX-License-Identifier: MIT

package com.hp.jipp.model

import com.hp.jipp.encoding.Keyword
import com.hp.jipp.encoding.KeywordType

/** A keyword attribute describing the action by which a printer may identify itself. */
class IdentifyAction(override val name: String) : Keyword() {

    companion object {
        @JvmField val display = IdentifyAction("display")
        @JvmField val flash = IdentifyAction("flash")
        @JvmField val sound = IdentifyAction("sound")
        @JvmField val speak = IdentifyAction("speak")

        val ENCODER = KeywordType.encoderOf(IdentifyAction::class.java) { IdentifyAction(it) }

        /** Return a new [KeywordType] for a type with the specified name */
        @JvmStatic operator fun invoke(name: String) = KeywordType(ENCODER, name)
    }
}
